### PR TITLE
[FLINK-19233][python] Support distinct and filter keywords on Python UDAF

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request supports distinct and filter keywords on Python UDAF.*

## Brief change log

  - *Support distinct keyword on Python UDAF.*
  - *support filter keyword on Python UDAF.*

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (yes)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-python/pyflink/table/tests/test_aggregate.py [hunk 1]
```diff
@@ -358,6 +386,48 @@ def test_map_view_iterate(self):
                 ["Hi,Hi2,Hi3", "1,2,3", "Hi:3,Hi2:2,Hi3:1", 3, "hi"]],
                 columns=['a', 'b', 'c', 'd', 'e']))
 
+    def test_distinct_and_filter(self):
+        self.t_env.create_temporary_system_function(
+            "concat",
+            ConcatAggregateFunction())
+        self.t_env.get_config().get_configuration().set_string(
```
Comment (dianfu) [inline flink-python/pyflink/table/tests/test_aggregate.py:393]: What's the purpose to set these configuration explicitly in this test case?

File: flink-python/pyflink/fn_execution/operations.py [hunk 2]
```diff
@@ -285,23 +285,46 @@ def open(self):
     def generate_func(self, serialized_fn):
         user_defined_aggs = []
         input_extractors = []
+        filter_args = []
+        # stores which distinct view current function shared with
+        distinct_mapping = []
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/operations.py:290]:
```suggestion
        distinct_view_indexes = []
```

File: flink-python/pyflink/fn_execution/aggregate.py [hunk 3]
```diff
@@ -287,7 +287,10 @@ def __init__(self,
                  udfs: List[AggregateFunction],
                  input_extractors: List,
                  index_of_count_star: int,
-                 udf_data_view_specs: List[List[DataViewSpec]]):
+                 udf_data_view_specs: List[List[DataViewSpec]],
+                 filter_args: List[int],
+                 distinct_mapping: List[int],
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/aggregate.py:292]:
```suggestion
                 distinct_view_indexes: List[int],
```

File: flink-python/pyflink/fn_execution/aggregate.py [hunk 4]
```diff
@@ -287,7 +287,10 @@ def __init__(self,
...
+                 distinct_view_reuse_descriptors: Dict[int, Tuple['function', List[int]]]):
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/aggregate.py:293]:
```suggestion
                 distinct_view_descriptors: Dict[int, Tuple['function', List[int]]]):
```

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your comments! I have updated this PR.

**Review by dianfu — APPROVED** [review]
LGTM. The failed tests is not related to this PR and has been fixed in FLINK-19793.

**Pull request closed without merging**